scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb

```diff
@@ -0,0 +1,354 @@
+
+            if (idxToDelete == -1) {
+                return false;
+            } else {
+                LineItems.RemoveAt(idxToDelete);
+
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Deletes one <see ref="CartLineItem">Line Item </see> based on a <see ref="ProductVariant">ProductVariant </see>. If a line
+        /// item was deleted, <c>true</c> will be returned. If no line items were deleted, <c>false</c> will be returned.
+        /// </summary>
+        /// <param name="variant"><see ref="ProductVariant">variant </see> to provide the ID to delete a line item</param>
+        /// \code
+        /// // Example that deletes a line item based on a product variant
+        /// Cart cart = ShopifyBuy.Client().Cart();
+        ///
+        /// Debug.Log("Did delete? " + cart.LineItems.Delete(variant));
+        /// \endcode
+        public bool Delete(ProductVariant variant) {
+            return Delete(variant.id());
+        }
+
+        /// <summary>
+        /// Deletes one <see ref="CartLineItem">Line Item </see> based on a <see ref="Product">Product </see> and selected options.
+        /// If a line item was deleted, <c>true</c> will be returned. If no line item was deleted, <c>false</c> will be returned.
+        /// </summary>
+        /// <param name="product"><see ref="Product">product </see> whose options will be used to determine which line item is deleted</param>
+        /// <param name="selectedOptions">a Dictionary used to define user selected options</param>
+        /// \code
+        /// // Example that deletes a line item based on a product and selected options
+        /// Cart cart = ShopifyBuy.Client().Cart();
+        ///
+        /// Dictionary<string, string> selectedOptions = new Dictionary<string, string>() {
+        ///     {"Size", "Small"},
+        ///     {"Color", "Red"}
+        /// };
+        ///
+        /// Debug.Log("Did delete? " + cart.LineItems.Delete(product, selectedOptions));
+        /// \endcode
+        public bool Delete(Product product, Dictionary<string, string> selectedOptions) {
+            string variantId = VariantIdFromSelectedOptions(product, selectedOptions);
+
+            if (variantId == null) {
+                return false;
+            }
+
+            return Delete(variantId);
+        }
+
+        private string VariantIdFromSelectedOptions(Product product, Dictionary<string, string> selectedOptions) {
+            List<ProductVariant> variants = (List<ProductVariant>) product.variants();
+            string variantId = null;
+
+            foreach(ProductVariant variant in variants) {
+                List<SelectedOption> variantSelectedOptions = variant.selectedOptions();
+                variantId = variant.id();
+
+                if (variantSelectedOptions.Count != selectedOptions.Keys.Count) {
+                    variantId = null;
```

Comment:
Would it be more clear if we removed this line, and used affirmative conditions? 
```C++
foreach(ProductVariant variant in variants) {
    List<SelectedOption> variantSelectedOptions = variant.selectedOptions();

    if (variantSelectedOptions.Count == selectedOptions.Keys.Count) {

        variantId = variant.id()

	foreach(SelectedOption variantOption in variantSelectedOptions) {
	    string optionName = variantOption.name();
            string optionValue = variantOption.value();

     	    if (!selectedOptions.ContainsKey(optionName) || selectedOptions[optionName] != optionValue) {
                variantId = null;
                break;
     	    }
	}

	if (variantId != null) {
		break;
	}
    }
}
```